torchx/examples/mnist.exs

```diff
@@ -0,0 +1,163 @@
+defmodule MNIST do
+  import Nx.Defn
+
+
+  defn init_random_params do
+    w1 = Nx.random_normal({784, 128}, 0.0, 0.1, names: [:input, :layer])
+    b1 = Nx.random_normal({128}, 0.0, 0.1, names: [:layer])
+    w2 = Nx.random_normal({128, 10}, 0.0, 0.1, names: [:layer, :output])
+    b2 = Nx.random_normal({10}, 0.0, 0.1, names: [:output])
+    {w1, b1, w2, b2}
+  end
+
+  defn softmax(logits) do
+    Nx.exp(logits) / Nx.sum(Nx.exp(logits), axes: [:output], keep_axes: true)
+  end
+
+  defn predict({w1, b1, w2, b2}, batch) do
+    batch
+    |> Nx.dot(w1)
+    |> Nx.add(b1)
+    |> Nx.logistic()
+    |> Nx.dot(w2)
+    |> Nx.add(b2)
+    |> softmax()
+  end
+
+  defn accuracy({w1, b1, w2, b2}, batch_images, batch_labels) do
+    Nx.mean(
+      Nx.equal(
+        Nx.argmax(Nx.as_type(batch_labels, {:s, 8}), axis: :output),
```

Comment:
We don't need this conversion here as argmax should work without casting.

Reply:
PyTorch does not implement argmax for unsigned byte.
Compilation fails without this type cast:

```
== Compilation error in file examples/mnist.exs ==
** (RuntimeError) Torchx: "argmax_cpu" not implemented for 'Bool' in NIF.argmax/3
```

Reply:
I see, thank you! My suggestion is for then to convert them to s8 before we call the neural network! I think this is for labels, which could be s8 from the beginning.